Add workflow to also run tests on a Debian container

dato opened the pull request:
This runs tests against Debian stable (bullseye), and testing (bookworm).

-----
I'm having a weird segmentation fault problem on my machine, so I ended up wanting to make sure that things work in vanilla Debian (they seemingly do, so my error must be elsewhere).

I had said a few days ago that I wanted this check anyway, to ensure we can build/run against the pre-compiled librocksdb Debian packages.

N.B.: At the moment both bullseye and bookworm have 6.11, but 6.25 is in experimental, so it should make it to bookworm at some point..

dato committed:
Add workflow to also run tests on a Debian container

This runs tests against Debian stable (bullseye) and, once rocksdb
6.25 makes it out of experimental, against bookworm too (testing).

NightTsarina commented on .github/workflows/build.yml:
```diff
@@ -2,7 +2,10 @@
+  pull_request:
   push:
+    branches:
+      - main
```
Isn't this too restrictive? Seems to be that this workflow is the most useful one to be run on every commit, and it is reasonably fast.
And also, if we are restricting on branches, it might be good to add a predefined branch for tests/development like you did on the debian one? Something like:
```suggestion
      - main
      - workflows/*
```

NightTsarina commented on .github/workflows/debian.yml:
```diff
@@ -0,0 +1,42 @@
+  push:
+    branches:
+      - main
+      - workflows/debian  # For debugging.
```
like I was suggesting above, maybe leave in all workflows a `workflows/*` condition for debugging/devel?

dato replied:
Ah, yes. The above was for debugging the _workflow_ itself, but you're right it would be useful to have a place to push where the test will be run.

How about `force_ci/debian/**` and `force_ci/all/**`? (And of course `force_ci/build/**` and `force_ci/all/**` for build.yml.)

NightTsarina replied:
Sounds great to me!

NightTsarina reviewed (approved):
LGTM. I left a couple of comments/suggestions, but those can be left for later if you prefer.

dato committed:
Specify force_ci/** branch pattern that can force a workflow to run

The previous commit restricting build.yml to only run on 'main'
(same as newly-introduced debian.yml). But it's useful to be able
to see CI runs, during development _before_ opening a pull request.

A branch pattern is added so that forcing CI to run is possible.
